atlasdb-workload-server/src/test/java/com/palantir/atlasdb/workload/invariant/SerializableInvariantTest.java

```diff
@@ -0,0 +1,149 @@
+        List<WitnessedTransaction> transactions = new WitnessedTransactionsBuilder("table")
+                .startTransaction()
+                .read(5, 10)
+                .write(5, 10, 15)
+                .read(5, 10, 15)
+                .endTransaction()
+                .build();
+        WorkflowHistory workflowHistory = ImmutableWorkflowHistory.builder()
+                .history(transactions)
+                .transactionStore(readableTransactionStore)
+                .build();
+        SerializableInvariant.INSTANCE.accept(workflowHistory, invalidTransactions::set);
+        assertThat(invalidTransactions.get()).isEmpty();
+    }
+
+    @Test
+    public void noInvalidTransactionsWhenSerializable() {
+        AtomicReference<List<InvalidWitnessedTransaction>> invalidTransactions = new AtomicReference<>();
+        List<WitnessedTransaction> transactions = new WitnessedTransactionsBuilder("table")
+                .startTransaction()
+                .write(5, 10, 15)
+                .endTransaction()
+                .startTransaction()
+                .read(5, 10, 15)
+                .write(23, 1, 20)
+                .write(7, 15, 14)
+                .endTransaction()
+                .startTransaction()
+                .read(5, 10, 15)
+                .read(23, 1, 20)
+                .read(7, 15, 14)
+                .endTransaction()
+                .build();
+        WorkflowHistory workflowHistory = ImmutableWorkflowHistory.builder()
+                .history(transactions)
+                .transactionStore(readableTransactionStore)
+                .build();
+        SerializableInvariant.INSTANCE.accept(workflowHistory, invalidTransactions::set);
+        assertThat(invalidTransactions.get()).isEmpty();
+    }
+
+    @Test
+    public void catchesWriteSkew() {
+        AtomicReference<List<InvalidWitnessedTransaction>> invalidTransactions = new AtomicReference<>();
+        List<WitnessedTransaction> transactions = new WitnessedTransactionsBuilder("table")
+                .startTransaction()
+                .write(5, 10, 15)
+                .write(6, 10, 15)
+                .endTransaction()
+                .startTransaction()
+                .read(5, 10, 15)
+                .read(6, 10, 15)
+                .write(5, 10, 0)
+                .endTransaction()
+                .startTransaction()
+                .read(5, 10, 15)
+                .read(6, 10, 15)
+                .write(6, 10, 0)
+                .endTransaction()
+                .build();
+        WorkflowHistory workflowHistory = ImmutableWorkflowHistory.builder()
+                .history(transactions)
+                .transactionStore(readableTransactionStore)
+                .build();
+        SerializableInvariant.INSTANCE.accept(workflowHistory, invalidTransactions::set);
+        assertThat(invalidTransactions).hasValueSatisfying(invalidWitnessedTransactions -> {
```

Comment:
We can probably avoid some of the logic here, by using just a direct `List` and `addAll` in the consumer passed to the invariant?

Reply:
this is a lot cleaner 🧼 updated everything to use this instead